package/win32/CMakeLists.txt

```diff
@@ -24,66 +24,130 @@ endif()
-
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.R' ''
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.RData' ''
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.rda' ''
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.Rd' ''
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.Rnw' ''
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.Rmd' ''
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.Rmarkdown' ''
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.qmd' ''
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.Rhtml' ''
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.Rpres' ''
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.Rproj' ''
-   WriteRegStr HKCR 'Applications\\\\rstudio.exe\\\\SupportedTypes' '.rdprsp' ''
-
-   WriteRegStr HKCR '.R\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.RData\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.rda\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.Rd\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.Rnw\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.Rmd\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.Rmarkdown\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.qmd\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.Rhtml\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.Rpres\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.Rproj\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.tex\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.md\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.mdtxt\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.markdown\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.htm\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.html\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.css\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.js\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.c\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.cpp\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.h\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.hpp\\\\OpenWithList\\\\rstudio.exe' '' ''
-   WriteRegStr HKCR '.rdprsp\\\\OpenWithList\\\\rstudio.exe' '' '' 
-
-   WriteRegStr HKCR '.Rproj' '' 'RStudio.Rproj'
-   WriteRegStr HKCR 'RStudio.Rproj' '' 'R Project'
-   WriteRegStr HKCR 'RStudio.Rproj\\\\DefaultIcon' '' '$INSTDIR\\\\bin\\\\rstudio.exe,-2'
-   WriteRegStr HKCR 'RStudio.Rproj\\\\\\\\shell\\\\open\\\\command' '' '$INSTDIR\\\\bin\\\\rstudio.exe \\\"%1\\\"'
-   
-   System::Call 'Shell32::SHChangeNotify(i 0x8000000, i 0, i 0, i 0)'
-")
+if(RSTUDIO_ELECTRON)
+
+   set(CPACK_NSIS_EXTRA_INSTALL_COMMANDS "
```

Comment:
If you want to avoid the need to escape slashes and embedded quotes, you could also try using a bracket argument, delimited with `[=[ <... > ]=]`. Sort of the CMake way of making a multi-line string.

Reply:
Just indentation changes here; this is long-standing existing code so going for minimal changes. But yes the newer cmake syntax would make this look less icky.